proposals/lockout-violation-detection.md

```diff
@@ -0,0 +1,96 @@
+### Lockout Violation Detection
+
+Assume:
+1. A database that we will store vote information ingested from
+gossip/turbine, which in turn will be consulted to detect lockout violations
+as follows.
+2. First we track all SlotHashes on the canonical/rooted fork in the database
+for the last epoch.
+
+The following sections will go over the possible lockout violations that will
+be detected and punished.
```

Comment:
> Who detects the violations and who punishes bad guys? Are these the same program?

Anybody that detects the violation can submit a proof to the network, which will then punish the offender. Exactly how this happens will be covered in a different proposal

> Say the world somehow got stuck and everyone needs to restart from a negotiated root (which might mean lockout violation), would this detection program comes in and bite us?

Yeah it would if we don't factor that into the design :) We can probably tag votes with an identifier corresponding to restarts, such that votes across restarts aren't slashable